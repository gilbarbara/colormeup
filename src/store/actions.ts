export { alertHide, alertShow } from './slices/alerts';
export { toggleSidebar } from './slices/app';
export { setColor, setOptions } from './slices/color';
export { removeColor, resetUserData, saveColor, setUserOptions } from './slices/user';
