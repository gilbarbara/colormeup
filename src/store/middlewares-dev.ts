import { Middleware } from '@reduxjs/toolkit';
import invariant from 'redux-immutable-state-invariant';
import { createLogger } from 'redux-logger';

export default [invariant(), createLogger({ collapsed: true })] as Middleware[];
