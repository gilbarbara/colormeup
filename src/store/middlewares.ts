import { Middleware } from 'redux';

const middlewares: Middleware[] = [];

if (process.env.NODE_ENV === 'development') {
  // eslint-disable-next-line @typescript-eslint/no-require-imports
  const devMiddlewares = require('./middlewares-dev').default;

  middlewares.push(...devMiddlewares);
}

export default middlewares;
