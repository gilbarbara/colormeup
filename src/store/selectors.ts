import { createSelector } from '@reduxjs/toolkit';

import { RootState } from '~/types';

const selectSelf = (state: RootState) => state;

export const selectApp = createSelector(selectSelf, state => state.app);

export const selectColor = createSelector(selectSelf, state => state.color);

export const selectUser = createSelector(selectSelf, state => state.user);
