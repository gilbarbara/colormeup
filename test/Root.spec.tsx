import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';

import Root from '~/Root';
import { store } from '~/store';

vi.mock('react-helmet-async', () => ({
  Helmet: ({ children }) => <div>{children}</div>,
  HelmetProvider: () => vi.fn(),
}));

describe('Root', () => {
  render(<Root />, { wrapper: ({ children }) => <Provider store={store}>{children}</Provider> });

  it('should render properly', () => {
    expect(screen.getByTestId('Root')).toMatchSnapshot();
  });
});
