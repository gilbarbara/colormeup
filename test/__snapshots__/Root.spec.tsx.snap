// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root > should render properly 1`] = `
<main
  class="css-hi6ly3"
  data-component-name="Root"
>
  <div />
  <div
    class="_main_75e16f css-x0gcdx"
    data-component-name="Sidebar"
  >
    <div
      class="css-1yyql7d"
      data-component-name="Info"
    >
      <div
        class="css-vm0sqq"
        data-component-name="CopyHEX"
      >
        <div
          class="css-1pvzks6"
        >
          <pre
            class="css-dg8sf6"
          >
            <span
              class="css-onif9v"
              data-component-name="Text"
            >
              #ff0044
            </span>
          </pre>
        </div>
        <span
          class="css-n92ui6"
          data-component-name="CopyToClipboard"
        >
          <div
            aria-label="Click to copy"
            class="css-9479jc"
            data-component-name="Tooltip"
            role="tooltip"
          >
            <span
              class="css-1sm3hig"
              data-component-name="Icon"
              name="copy"
            />
          </div>
        </span>
      </div>
      <div
        class="css-vm0sqq"
        data-component-name="CopyRGB"
      >
        <div
          class="css-1pvzks6"
        >
          <pre
            class="css-dg8sf6"
          >
            <span
              class="css-onif9v"
              data-component-name="Text"
            >
              rgb(255, 0, 68)
            </span>
          </pre>
        </div>
        <span
          class="css-n92ui6"
          data-component-name="CopyToClipboard"
        >
          <div
            aria-label="Click to copy"
            class="css-9479jc"
            data-component-name="Tooltip"
            role="tooltip"
          >
            <span
              class="css-1sm3hig"
              data-component-name="Icon"
              name="copy"
            />
          </div>
        </span>
      </div>
      <div
        class="css-vm0sqq"
        data-component-name="CopyHSL"
      >
        <div
          class="css-1pvzks6"
        >
          <pre
            class="css-dg8sf6"
          >
            <span
              class="css-onif9v"
              data-component-name="Text"
            >
              hsl(344, 100%, 50%)
            </span>
          </pre>
        </div>
        <span
          class="css-n92ui6"
          data-component-name="CopyToClipboard"
        >
          <div
            aria-label="Click to copy"
            class="css-9479jc"
            data-component-name="Tooltip"
            role="tooltip"
          >
            <span
              class="css-1sm3hig"
              data-component-name="Icon"
              name="copy"
            />
          </div>
        </span>
      </div>
      <span
        class="css-1t7vy3y"
        data-component-name="FlexInline"
      >
        <span
          class="css-onif9v"
          data-component-name="Text"
        >
          color
        </span>
         
        #ffffff
      </span>
      <span
        class="css-1t7vy3y"
        data-component-name="FlexInline"
      >
        <span
          class="css-onif9v"
          data-component-name="Text"
        >
          chroma
        </span>
         
        1
      </span>
      <span
        class="css-1t7vy3y"
        data-component-name="FlexInline"
      >
        <span
          class="css-onif9v"
          data-component-name="Text"
        >
          luminance
        </span>
         
        0.2168
      </span>
    </div>
    <div>
      <p
        class="css-maf0tk"
        data-component-name="Paragraph"
      >
        <span
          class="css-3bqh9n"
          data-component-name="Text"
        >
          <span
            class="css-11ckkx5"
            data-component-name="Icon"
            name="heart"
          />
           favorites
        </span>
      </p>
      <div
        class="css-u4e6ve"
        data-component-name="Flex"
      >
        No favorites yet!
      </div>
    </div>
    <div
      class="_help_75e16f"
    >
      <button
        class="css-1y6ob7l"
        data-component-name="ButtonUnstyled"
        type="button"
      >
        <span
          class="css-11ckkx5"
          data-component-name="Icon"
          name="info-o"
        />
        <span
          class="css-3bqh9n"
          data-component-name="Text"
        >
          help
        </span>
      </button>
      <div
        class="css-15onlen"
        data-component-name="Flex"
      >
        <p
          class="css-1xr4sq5"
          data-component-name="Paragraph"
        >
          <span
            class="css-1sl0vs1"
            data-component-name="Text"
          >
            Know your colors!
          </span>
        </p>
        <p
          class="css-1d6m1sj"
          data-component-name="Paragraph"
        >
          <span
            class="css-onif9v"
            data-component-name="Text"
          >
            colormeup is a tool to inspect colors and play with its many variations in HSL or RGB.
          </span>
        </p>
        <span
          class="css-12wyrx8"
          data-component-name="FlexInline"
        >
          <span
            class="css-1vb77dn"
            data-component-name="Icon"
            name="layout-grid"
          />
          the number of colors
        </span>
        <span
          class="css-12wyrx8"
          data-component-name="FlexInline"
        >
          <span
            class="css-1vb77dn"
            data-component-name="Icon"
            name="sync"
          />
          generate a new color
        </span>
        <span
          class="css-12wyrx8"
          data-component-name="FlexInline"
        >
          <span
            class="css-1vb77dn"
            data-component-name="Icon"
            name="heart"
          />
           save to your favorites
        </span>
      </div>
    </div>
  </div>
  <div
    class="_main_053da1"
    data-component-name="Header"
    style="background-color: rgb(255, 0, 68); border-color: #b30030;"
  >
    <div
      class="_content_053da1"
    >
      <div
        class="_main_16f863"
        data-component-name="Logo"
      >
        <svg
          enable-background="new 0 0 400 70"
          preserveAspectRatio="xMinYMin meet"
          version="1.1"
          viewBox="0 0 400 70"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M44.3 38.2c-.4 3.4-1.3 6.3-2.8 8.7-1.5 2.5-3.3 4.5-5.5 6.1-2.1 1.6-4.5 2.8-7.1 3.5s-5.2 1-7.9.9c-2.7-.2-5.3-.7-7.8-1.7s-4.8-2.4-6.7-4.3c-2-1.9-3.5-4.2-4.7-6.9-1.2-2.8-1.8-6-1.8-9.6 0-3.6.6-6.8 1.8-9.5 1.2-2.7 2.8-5 4.7-6.9 2-1.9 4.2-3.3 6.7-4.3s5.1-1.6 7.8-1.7c2.7-.1 5.3.2 7.9.9s5 1.9 7.1 3.5c2.1 1.6 4 3.6 5.4 6.1 1.5 2.5 2.4 5.4 2.8 8.7H31.6c-.5-1.7-1.3-3.2-2.4-4.3-1.1-1.1-2.4-1.9-3.8-2.4s-2.9-.6-4.4-.4c-1.5.2-2.9.7-4.2 1.5-1.3.8-2.3 2-3.1 3.5-.8 1.5-1.2 3.3-1.2 5.4 0 2.1.4 3.9 1.2 5.4.8 1.5 1.9 2.7 3.1 3.5 1.3.8 2.7 1.4 4.2 1.5 1.5.2 3 .1 4.4-.4 1.4-.5 2.7-1.3 3.8-2.4 1.1-1.1 1.9-2.6 2.4-4.3h12.7zM92.1 34.9c0 3.8-.7 7.1-2 9.9-1.3 2.8-3 5.2-5.2 7.1-2.1 1.9-4.6 3.3-7.3 4.3-2.7.9-5.5 1.4-8.3 1.4-2.8 0-5.6-.5-8.3-1.4-2.7-.9-5.1-2.4-7.3-4.3-2.1-1.9-3.9-4.3-5.2-7.1-1.3-2.8-2-6.1-2-9.9 0-3.8.6-7.1 2-10 1.3-2.8 3-5.2 5.2-7.1 2.1-1.9 4.6-3.3 7.3-4.3 2.7-.9 5.5-1.4 8.3-1.4 2.8 0 5.6.5 8.3 1.4 2.7.9 5.1 2.4 7.3 4.3 2.1 1.9 3.9 4.3 5.2 7.1 1.3 2.9 2 6.3 2 10zm-32.7 0c0 2.3.5 4.3 1.5 5.8s2.2 2.7 3.7 3.5c1.5.8 3.1 1.2 4.8 1.2s3.3-.4 4.8-1.2c1.5-.8 2.7-2 3.7-3.5s1.5-3.5 1.5-5.8-.5-4.3-1.5-5.9c-1-1.6-2.2-2.7-3.7-3.5-1.5-.8-3.1-1.2-4.8-1.2s-3.3.4-4.8 1.2c-1.5.8-2.7 2-3.7 3.5-1 1.7-1.5 3.6-1.5 5.9zM108.7 0v56.9h-13V0h13zM157.3 34.9c0 3.8-.7 7.1-2 9.9-1.3 2.8-3 5.2-5.2 7.1-2.1 1.9-4.6 3.3-7.3 4.3-2.7.9-5.5 1.4-8.3 1.4-2.8 0-5.6-.5-8.3-1.4-2.7-.9-5.1-2.4-7.3-4.3-2.1-1.9-3.9-4.3-5.2-7.1-1.3-2.8-2-6.1-2-9.9 0-3.8.6-7.1 2-10 1.3-2.8 3-5.2 5.2-7.1 2.1-1.9 4.6-3.3 7.3-4.3 2.7-.9 5.5-1.4 8.3-1.4 2.8 0 5.6.5 8.3 1.4 2.7.9 5.1 2.4 7.3 4.3 2.1 1.9 3.9 4.3 5.2 7.1 1.3 2.9 2 6.3 2 10zm-32.7 0c0 2.3.5 4.3 1.5 5.8s2.2 2.7 3.7 3.5c1.5.8 3.1 1.2 4.8 1.2s3.3-.4 4.8-1.2c1.5-.8 2.7-2 3.7-3.5s1.5-3.5 1.5-5.8-.5-4.3-1.5-5.9c-1-1.6-2.2-2.7-3.7-3.5-1.5-.8-3.1-1.2-4.8-1.2s-3.3.4-4.8 1.2c-1.5.8-2.7 2-3.7 3.5-1 1.7-1.5 3.6-1.5 5.9zM173.5 32.5v24.4h-12.8v-44h11.5l.5 2.9c2.1-2 4.3-3.1 6.7-3.6 2.4-.4 5-.3 7.8.4l-.2 12c-1.4-.4-2.8-.6-4.4-.6-1.6 0-3.1.3-4.4.9-1.4.6-2.5 1.5-3.4 2.7-.8 1.4-1.3 3-1.3 4.9z"
            fill="#c4ff00"
            fill-opacity="1"
          />
          <path
            d="M190.8 12.9h12.9v4.4c1.6-2 3.4-3.3 5.6-4 2.2-.7 4.4-1.1 6.8-1 2.5.1 4.6.7 6.3 1.7s3 2.4 4 4.1c1.5-2.1 3.5-3.6 5.9-4.5 2.4-.9 4.9-1.4 7.5-1.3 2.1.1 4 .5 5.8 1.3 1.8.8 3.4 1.9 4.8 3.5s2.4 3.6 3.2 6.1c.8 2.5 1.1 5.4 1.1 8.9v24.8h-12.9V32.2c0-2.7-.5-4.5-1.6-5.6-1.1-1.1-2.4-1.7-3.8-1.7-.8 0-1.6.1-2.4.4-.8.3-1.5.7-2.2 1.3s-1.2 1.3-1.6 2.2c-.4.9-.7 1.9-.7 3.1v25h-13V32.2c0-2.7-.5-4.5-1.6-5.6-1.1-1.1-2.4-1.7-3.8-1.7-.8 0-1.6.1-2.4.4-.8.3-1.5.7-2.2 1.3-.7.6-1.2 1.3-1.6 2.2s-.6 2-.6 3.2v24.9h-13.6v-44zM303.6 34.8c0 1.7-.2 3.4-.5 5h-31.5c.6 1.7 1.7 3.1 3.1 4.1 1.4 1 3 1.6 4.6 1.9 1.7.3 3.3.2 5-.3 1.7-.5 3-1.3 4.1-2.5h13.9c-1.2 3.6-3.2 6.6-5.9 8.8-2.7 2.2-5.6 3.8-8.9 4.7-3.3.9-6.7 1.1-10.1.7-3.5-.5-6.6-1.6-9.5-3.4-2.8-1.8-5.2-4.3-7-7.5-1.8-3.2-2.8-7-2.8-11.5 0-3.7.7-7 2-9.8 1.3-2.8 3-5.2 5.2-7.1 2.1-1.9 4.6-3.3 7.3-4.2 2.7-.9 5.5-1.4 8.3-1.4 2.8 0 5.6.5 8.3 1.4 2.7.9 5.1 2.3 7.3 4.2 2.1 1.9 3.9 4.3 5.2 7.1 1.3 2.8 1.9 6 1.9 9.8zm-14-5c-.6-2.2-1.7-3.8-3.4-4.9-1.7-1-3.5-1.5-5.6-1.5-1.9 0-3.6.5-5.2 1.6s-2.7 2.7-3.4 4.8h17.6z"
            fill="#00ffbb"
            fill-opacity="1"
          />
          <path
            d="M348.5 56.9h-12.9v-4.5c-1.6 2-3.4 3.4-5.6 4.1-2.2.7-4.4 1-6.8 1-3.1 0-5.7-.6-7.8-1.7-2-1.1-3.7-2.6-4.9-4.4-1.2-1.8-2.1-4-2.5-6.5-.5-2.5-.7-5.1-.7-7.8V12.9h12.9V37c0 2.2.4 3.9 1.1 5.2.7 1.3 1.6 2.3 2.7 2.8 1.1.6 2.3.8 3.5.7 1.3-.1 2.5-.5 3.5-1.3 1.1-.7 2-1.7 2.7-3s1.1-2.9 1.1-4.7V12.9h13.6v44zM377.4 57.6c-2.2 0-4.2-.4-6.2-1.1-2-.8-3.7-2-5.2-3.7v16.9h-12.7V12.9H366v4.4c1.5-1.9 3.3-3.2 5.3-3.9 2-.7 4.1-1.1 6.2-1.1 3.5 0 6.5.7 9.2 2 2.7 1.3 4.9 3 6.7 5.2 1.8 2.1 3.1 4.6 4 7.4.9 2.8 1.3 5.7 1.3 8.8-.1 3-.5 5.9-1.4 8.5-.9 2.7-2.2 5-4 6.9-1.8 2-4 3.5-6.6 4.7-2.6 1.2-5.7 1.8-9.3 1.8zm-1.4-33c-1.4 0-2.7.3-3.9.8-1.2.5-2.2 1.3-3.2 2.3-.9 1-1.6 2.1-2.2 3.4-.5 1.3-.8 2.6-.8 4.1s.3 2.8.8 4.1c.5 1.3 1.2 2.4 2.2 3.4.9.9 2 1.7 3.2 2.3 1.2.6 2.5.9 3.9.9 1.6 0 3-.3 4.3-.9 1.2-.6 2.2-1.4 3-2.4.8-1 1.4-2.1 1.9-3.4.4-1.3.7-2.6.7-3.9s-.2-2.6-.6-3.9-1-2.5-1.8-3.5-1.8-1.8-3.1-2.4c-1.3-.6-2.7-.9-4.4-.9z"
            fill="#3b00ff"
            fill-opacity="1"
          />
        </svg>
      </div>
      <div
        class="css-1j3y15i"
        data-component-name="Flex"
      >
        <div
          class="_options_053da1"
          data-component-name="HeaderOptions"
        >
          <div
            class="_optionsGroup_053da1"
            data-component-name="HeaderOptionsGroup"
          >
            <span>
              <button
                class="css-1y6ob7l"
                data-component-name="ButtonUnstyled"
                title="Randomize Color"
                type="button"
              >
                <span
                  class="css-e9rmiv"
                  data-component-name="Icon"
                  name="sync"
                />
              </button>
            </span>
            <input
              name="color"
              tabindex="0"
              type="text"
              value="#ff0044"
            />
            <span>
              <span
                class="css-p9e7si"
                data-component-name="CopyToClipboard"
              >
                <div
                  aria-label="Click to copy"
                  class="css-9479jc"
                  data-component-name="Tooltip"
                  role="tooltip"
                >
                  <span
                    class="css-e9rmiv"
                    data-component-name="Icon"
                    name="copy"
                  />
                </div>
              </span>
            </span>
            <span>
              <button
                class="css-1y6ob7l"
                data-component-name="ButtonUnstyled"
                title="Add to Favorites"
                type="button"
              >
                <span
                  class="css-e9rmiv"
                  data-component-name="Icon"
                  name="heart"
                />
              </button>
            </span>
          </div>
        </div>
        <div
          class="_main_d90600"
          data-component-name="Sliders"
        >
          <div
            class="_models_d90600"
            data-component-name="SlidersModels"
          >
            <button
              class="_modelBtn_d90600 _active_d90600"
              data-model="hsl"
              type="button"
            >
              HSL
            </button>
            <button
              class="_modelBtn_d90600"
              data-model="rgb"
              type="button"
            >
              RGB
            </button>
          </div>
          <div
            class="_sliders_d90600"
          >
            <div
              class="_sliderMain_d90600"
              data-component-name="Slider"
            >
              <div
                class="_sliderName_d90600"
              >
                Hue
              </div>
              <div
                data-type="h"
                style="box-sizing: border-box; display: inline-block; padding: 6px; transition: height 0.4s, width 0.4s; height: 22px; width: 100%;"
              >
                <div
                  role="presentation"
                  style="background-color: rgb(204, 204, 204); border-radius: 8px; box-sizing: border-box; height: 10px; position: relative; width: 100%;"
                >
                  <div
                    style="width: 95.55555555555556%; background-color: rgb(170, 170, 170); border-radius: 8px; position: absolute; height: 100%; top: 0px;"
                  />
                  <div
                    role="presentation"
                    style="box-sizing: border-box; height: 10px; position: absolute; transition: height 0.4s, width 0.4s; width: 20px; left: 95.55555555555556%; bottom: 0%;"
                  >
                    <span
                      aria-label="slider handle"
                      aria-orientation="horizontal"
                      aria-valuemax="360"
                      aria-valuemin="0"
                      aria-valuenow="344"
                      role="slider"
                      style="background-color: rgb(136, 136, 136); border: 0px; border-radius: 8px; box-sizing: border-box; display: block; position: absolute; transition: height 0.4s, width 0.4s; height: 12px; left: -6px; top: -1px; width: 12px;"
                      tabindex="0"
                    />
                  </div>
                </div>
              </div>
              <div
                class="_main_c9ae10"
                role="presentation"
              >
                <input
                  class="_input_c9ae10"
                  name="h"
                  type="text"
                  value="344"
                />
                <div
                  class="_buttons_c9ae10"
                >
                  <button
                    class="_button_c9ae10"
                    data-direction="up"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-up"
                    />
                  </button>
                  <button
                    class="_button_c9ae10"
                    data-direction="down"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-down"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="_sliderMain_d90600"
              data-component-name="Slider"
            >
              <div
                class="_sliderName_d90600"
              >
                Saturation
              </div>
              <div
                data-type="s"
                style="box-sizing: border-box; display: inline-block; padding: 6px; transition: height 0.4s, width 0.4s; height: 22px; width: 100%;"
              >
                <div
                  role="presentation"
                  style="background-color: rgb(204, 204, 204); border-radius: 8px; box-sizing: border-box; height: 10px; position: relative; width: 100%;"
                >
                  <div
                    style="width: 100%; background-color: rgb(170, 170, 170); border-radius: 8px; position: absolute; height: 100%; top: 0px;"
                  />
                  <div
                    role="presentation"
                    style="box-sizing: border-box; height: 10px; position: absolute; transition: height 0.4s, width 0.4s; width: 20px; left: 100%; bottom: 0%;"
                  >
                    <span
                      aria-label="slider handle"
                      aria-orientation="horizontal"
                      aria-valuemax="100"
                      aria-valuemin="0"
                      aria-valuenow="100"
                      role="slider"
                      style="background-color: rgb(136, 136, 136); border: 0px; border-radius: 8px; box-sizing: border-box; display: block; position: absolute; transition: height 0.4s, width 0.4s; height: 12px; left: -6px; top: -1px; width: 12px;"
                      tabindex="0"
                    />
                  </div>
                </div>
              </div>
              <div
                class="_main_c9ae10"
                role="presentation"
              >
                <input
                  class="_input_c9ae10"
                  name="s"
                  type="text"
                  value="100"
                />
                <div
                  class="_buttons_c9ae10"
                >
                  <button
                    class="_button_c9ae10"
                    data-direction="up"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-up"
                    />
                  </button>
                  <button
                    class="_button_c9ae10"
                    data-direction="down"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-down"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="_sliderMain_d90600"
              data-component-name="Slider"
            >
              <div
                class="_sliderName_d90600"
              >
                Lightness
              </div>
              <div
                data-type="l"
                style="box-sizing: border-box; display: inline-block; padding: 6px; transition: height 0.4s, width 0.4s; height: 22px; width: 100%;"
              >
                <div
                  role="presentation"
                  style="background-color: rgb(204, 204, 204); border-radius: 8px; box-sizing: border-box; height: 10px; position: relative; width: 100%;"
                >
                  <div
                    style="width: 50%; background-color: rgb(170, 170, 170); border-radius: 8px; position: absolute; height: 100%; top: 0px;"
                  />
                  <div
                    role="presentation"
                    style="box-sizing: border-box; height: 10px; position: absolute; transition: height 0.4s, width 0.4s; width: 20px; left: 50%; bottom: 0%;"
                  >
                    <span
                      aria-label="slider handle"
                      aria-orientation="horizontal"
                      aria-valuemax="100"
                      aria-valuemin="0"
                      aria-valuenow="50"
                      role="slider"
                      style="background-color: rgb(136, 136, 136); border: 0px; border-radius: 8px; box-sizing: border-box; display: block; position: absolute; transition: height 0.4s, width 0.4s; height: 12px; left: -6px; top: -1px; width: 12px;"
                      tabindex="0"
                    />
                  </div>
                </div>
              </div>
              <div
                class="_main_c9ae10"
                role="presentation"
              >
                <input
                  class="_input_c9ae10"
                  name="l"
                  type="text"
                  value="50"
                />
                <div
                  class="_buttons_c9ae10"
                >
                  <button
                    class="_button_c9ae10"
                    data-direction="up"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-up"
                    />
                  </button>
                  <button
                    class="_button_c9ae10"
                    data-direction="down"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-down"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="_selectors_053da1 css-18lcn0u"
          data-component-name="HeaderSelectors"
        >
          <div
            class="_selectors_053da1 css-1d5bbv1"
            data-component-name="SpacerItem"
          >
            <div
              aria-label="hsl"
              class="css-7ygwe"
              data-component-name="ButtonGroup"
            >
              <button
                class="css-nuj65q"
                data-component-name="Button"
                data-filled="true"
                data-type="h"
                type="button"
              >
                Hue
              </button>
              <button
                class="css-1jwlmwn"
                data-component-name="Button"
                data-filled="false"
                data-type="s"
                type="button"
              >
                Saturation
              </button>
              <button
                class="css-1jwlmwn"
                data-component-name="Button"
                data-filled="false"
                data-type="l"
                type="button"
              >
                Lightness
              </button>
            </div>
          </div>
          <div
            class="_selectors_053da1 css-1d5bbv1"
            data-component-name="SpacerItem"
          >
            <div
              aria-label="rgb"
              class="css-7ygwe"
              data-component-name="ButtonGroup"
            >
              <button
                class="css-1jwlmwn"
                data-component-name="Button"
                data-filled="false"
                data-type="r"
                type="button"
              >
                Red
              </button>
              <button
                class="css-1jwlmwn"
                data-component-name="Button"
                data-filled="false"
                data-type="g"
                type="button"
              >
                Green
              </button>
              <button
                class="css-1jwlmwn"
                data-component-name="Button"
                data-filled="false"
                data-type="b"
                type="button"
              >
                Blue
              </button>
            </div>
          </div>
          <div
            class="_selectors_053da1 css-1d5bbv1"
            data-component-name="SpacerItem"
          >
            <div
              class="css-3o3ltg"
              data-component-name="HeaderSteps"
            >
              <span
                class="css-1nsv1tr"
                data-component-name="Icon"
                name="layout-grid"
              />
              <div
                class="_main_c9ae10"
                role="presentation"
              >
                <input
                  class="_input_c9ae10"
                  name="steps"
                  type="text"
                  value="16"
                />
                <div
                  class="_buttons_c9ae10"
                >
                  <button
                    class="_button_c9ae10"
                    data-direction="up"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-up"
                    />
                  </button>
                  <button
                    class="_button_c9ae10"
                    data-direction="down"
                    type="button"
                  >
                    <span
                      class="css-pnao7i"
                      data-component-name="Icon"
                      name="chevron-down"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="_toggle_053da1"
        data-component-name="HeaderToggle"
      >
        <label
          aria-label="Toggle Menu"
          for="sidebar-toggle"
        >
          <input
            id="sidebar-toggle"
            type="checkbox"
          />
          <span
            style="color: rgb(196, 255, 0);"
          />
        </label>
      </div>
    </div>
  </div>
  <div
    class="_main_7a9327"
    data-component-name="Color"
  >
    <div
      class="_boxes_7a9327"
      data-component-name="ColorBoxes"
    >
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#ff0044"
        style="background-color: rgb(255, 0, 68);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: bold;"
        >
          #ff0044
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#ff1c00"
        style="background-color: rgb(255, 28, 0);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: normal;"
        >
          #ff1c00
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#ff7b00"
        style="background-color: rgb(255, 123, 0);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #ff7b00
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#ffdb00"
        style="background-color: rgb(255, 219, 0);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #ffdb00
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#c4ff00"
        style="background-color: rgb(196, 255, 0);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #c4ff00
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#64ff00"
        style="background-color: rgb(100, 255, 0);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #64ff00
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#04ff00"
        style="background-color: rgb(4, 255, 0);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #04ff00
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#00ff5b"
        style="background-color: rgb(0, 255, 91);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #00ff5b
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#00ffbb"
        style="background-color: rgb(0, 255, 187);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #00ffbb
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#00e3ff"
        style="background-color: rgb(0, 227, 255);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(0, 0, 0); font-weight: normal;"
        >
          #00e3ff
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#0084ff"
        style="background-color: rgb(0, 132, 255);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: normal;"
        >
          #0084ff
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#0024ff"
        style="background-color: rgb(0, 36, 255);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: normal;"
        >
          #0024ff
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#3b00ff"
        style="background-color: rgb(59, 0, 255);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: normal;"
        >
          #3b00ff
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#9b00ff"
        style="background-color: rgb(155, 0, 255);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: normal;"
        >
          #9b00ff
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#fb00ff"
        style="background-color: rgb(251, 0, 255);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: normal;"
        >
          #fb00ff
        </div>
      </a>
      <a
        class="_colorLink_7a9327"
        data-discover="true"
        href="/#ff00a4"
        style="background-color: rgb(255, 0, 164);"
      >
        <div
          class="_colorBox_7a9327"
          data-component-name="ColorBox"
          style="color: rgb(255, 255, 255); font-weight: normal;"
        >
          #ff00a4
        </div>
      </a>
    </div>
  </div>
  <footer
    class="css-1hchgaw"
    data-component-name="Footer"
  >
    <p
      class="css-1d6m1sj"
      data-component-name="Paragraph"
    >
      <span
        class="css-onif9v"
        data-component-name="Text"
      >
        © 2015 Gil Barbara
      </span>
    </p>
    <a
      class="css-165aw6i"
      data-component-name="Anchor"
      href="https://github.com/gilbarbara/colormeup"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        <span
          class="css-1g4cqre"
          data-component-name="Icon"
          name="github"
        />
      </span>
    </a>
  </footer>
  <div
    class="css-1cse04e"
    data-component-name="Overlay"
  />
  <div
    class="css-6v82yi"
    data-component-name="SystemAlerts"
  >
    <div
      class="css-w54ufa"
    >
      <div />
    </div>
    <div
      class="css-1f61f17"
    >
      <div />
    </div>
    <div
      class="css-i7fygl"
    >
      <div />
    </div>
    <div
      class="css-14e2mbs"
    >
      <div />
    </div>
  </div>
</main>
`;
